packages/avatars/src/styled/StyledStatusIndicator.ts

```diff
@@ -60,22 +62,26 @@ const sizeStyles = (props: IStatusIndicatorProps & ThemeProps<DefaultTheme>) =>
   `;
 };
 
-const colorStyles = (props: IStatusIndicatorProps & ThemeProps<DefaultTheme>) => {
-  const { theme, type, size, borderColor, surfaceColor } = props;
-
+const colorStyles = ({
+  theme,
+  $type,
+  $size,
+  $borderColor,
+  $surfaceColor
+}: IStatusIndicatorProps & ThemeProps<DefaultTheme>) => {
   let boxShadow = theme.shadows.sm(
-    surfaceColor ||
-      (type
-        ? getColorV8('background', 600 /* default shade */, theme)!
+    $surfaceColor ||
+      ($type
+        ? getColor({ variable: 'background.default', theme })
         : (theme.palette.white as string))
   );
 
-  if (size === xxs) {
+  if ($size === xxs) {
     boxShadow = boxShadow.replace(theme.shadowWidths.sm, '1px');
   }
```

Comment:
Potential performance improvement by setting `let boxShadow;` undefined and moving the `theme.shadows.sm` to a conditional `else` clause here.

Reply:
Not sure I'm seeing the gain. We calculate the box shadow either way, don't we? 🤔

Reply:
On second thought, I'm not actually entirely sure what you mean by moving the `theme.shadows.sm` to an `else` clause. Can you elaborate?

Reply:
Ah, I missed the `replace`. That's pretty odd treatment for our typical goto. I think the code should follow normative Garden conventions and look more like this...

```
const boxShadowColor = surfaceColor || getColor({ theme, variable: 'background.default' })
let boxShadow;

if (size === xxs) {
  boxShadow = theme.shadows.xs(boxShadowColor);
} else {
  boxShadow = theme.shadows.sm(boxShadowColor);
}
```

Stepping away from the theme like the current code does isn't ideal.

Reply:
Either that or make the `boxShadow` a simple ternary...

```
const boxShadow = size === xxs ? theme.shadows.xs(boxShadowColor) : theme.shadows.sm(boxShadowColor);
```

Reply:
That makes total sense! Thanks for clarifying.